### ISSUE TITLE
Unoptimized events in a special case related with escaping.

### ISSUE THREAD

**Issue by crlf0710** [raphlinus/pulldown-cmark#591]
Currently when parsing this:
```text
\***
```
Three events each with a single char is generated. Isn't it better expressed with one single event?

**Comment by marcusklaas** [comment]
Yep, that would be more efficient. I remember experimenting with logic to merge text events, and it was a lot faster for such examples, but it was making the general case a bit slower, so I decided against it.

If it can be done without such a trade-off, I'd be all for it!

**Comment by crlf0710** [comment]
I'm not very familiar with the internals of the code yet, i wonder if it is possible for someone to point me to some snippet in charge of this or some existing branch so i can get started hacking and experimenting with it :)

**Comment by equt** [comment]
Specifically, it's at https://github.com/raphlinus/pulldown-cmark/blob/4d45c33e0ca450044dfb030b072a41978d709c98/src/firstpass.rs#L540-L559
